dmscripts/supplier_data.py

```diff
@@ -0,0 +1,103 @@
+
+    def get_supplier_service_data(self, supplier_id):
+        """Given a supplier ID return a list of dictionaries for services related to framework."""
+        return self.client.find_draft_services_iter(supplier_id, framework=self.target_framework_slug)
+
+    def populate_data(self):
+        """Populate a dict with supplier data from the api."""
+        self.data = self.get_supplier_frameworks()
+        users = self.get_supplier_users()
+        for supplier_framework in self.data:
+            supplier_id = supplier_framework['supplierId']
+            supplier_framework['users'] = users.get(supplier_id, [])
+            supplier_framework['services'] = list(self.get_supplier_service_data(supplier_id))
+
+
+class SuccessfulSupplierContextForNotify(SupplierFrameworkData):
+    """Get the personalisation/ context for 'Your application result - if successful email'"""
+
+    STATUS_MAP = {'submitted': 'Successful', 'not-submitted': 'No application', 'failed': 'Unsuccessful'}
+
+    def __init__(self, client, target_framework_slug):
+        """Get the target framework to operate on and list the lots.
+
+        :param client: Instantiated api client
+        :param target_framework_slug: Framework to fetch data for
+        """
+        self.date_today = date.today().strftime(DISPLAY_DATE_FORMAT)
+        super(SuccessfulSupplierContextForNotify, self).__init__(client, target_framework_slug)
+
+        self.framework = client.get_framework(self.target_framework_slug)['frameworks']
+        self.framework_lots = [i['name'] for i in self.framework['lots']]
+
+    def get_users_peronalisations(self):
+        """Return {email_address: {personalisations}} for users eligible for the
+        'Your application result - if successful' email
+        """
+        output = {}
+        for supplier_framework in filter(lambda i: i['onFramework'], self.data):
+            for user in supplier_framework['users']:
+                output.update(self.get_user_personalisation(user, supplier_framework))
+        return output
+
+    def get_lot_dict(self, supplier_framework):
+        """Return a dict of lot status for each lot on the framework.
+        """
+        any_lots = False
+
+        lot_dict = OrderedDict((lot_name, 'No application') for lot_name in self.framework_lots)
+        for service in supplier_framework['services']:
+            status = service['status']
+            if status == 'submitted':
+                any_lots = True
+            lot_dict[service['lotName']] = self.STATUS_MAP[status]
+        if not any_lots:
+            return {}
+        return lot_dict
+
+    def get_user_personalisation(self, user, supplier_framework):
+        """Get dict of all info required by template given a user and framework."""
+        lot_dict = self.get_lot_dict(supplier_framework)
+        if not lot_dict:
+            return {}
+        lot_dict = dict(zip(
+            ['lot_%d' % i for i in range(1, len(self.framework_lots) + 1)],
```

Comment:
I'm not sure the ordering of the lots as they come out of the API is guaranteed, so numbering by the order in the dict seems like it might be inconsistent.  If we need the lots to be numbered then we should maybe just hard code the mapping of lot names here to be on the safe side.